lib/ansible/playbook/role/__init__.py

```diff
@@ -294,7 +294,7 @@ def _get_role_argspecs(self):
             if self._loader.path_exists(full_path):
                 # Note: _load_role_yaml() takes care of rebuilding the path.
                 argument_specs = self._load_role_yaml('meta', main='argument_specs')
-                return argument_specs.get('argument_specs', {})
+                return argument_specs.get('argument_specs', {}) if argument_specs is not None else {}
```

Comment:
This could probably be shortened

```suggestion
                return argument_specs.get('argument_specs', {}) or {}
```

Reply:
nvm, too early, no coffee

Reply:
to better cover all cases
```
try:
   return argument_specs.get('argument_specs')  or {}
except AttributeError
  return {]
```